google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -775,6 +776,15 @@ public void close() {
       if (lastException != null && isSessionNotFound(lastException)) {
         invalidateSession(this);
       } else {
+        if (lastException != null && isDatabaseNotFound(lastException)) {
```

Comment:
nits: use `else if`?

Reply:
That wouldn't work, as the else block also contains a non-conditional block underneath this `if`-block.